cli/azd/pkg/commands/pipeline/github_provider.go

```diff
@@ -358,17 +364,73 @@ func (p *GitHubCiProvider) configureConnection(
 	repoDetails *gitRepositoryDetails,
 	infraOptions provisioning.Options,
 	credentials json.RawMessage,
+	authType PipelineAuthType,
 	console input.Console) error {
 
+	// Federated Auth + Terraform is not a supported combination
+	if infraOptions.Provider == provisioning.Terraform {
+		// Throw error if Federated auth is explicitly requested
+		if authType == AuthTypeFederated {
+			return fmt.Errorf(
+				//nolint:lll
+				"Terraform does not support federated authentication. Service Principal with client ID and client secret must be used. %w",
+				ErrAuthNotSupported,
+			)
+		}
+
+		// If not explicitly set, show warning
```

Comment:
So, I don't understand how this `If not explicitly set` works.
I might be missing something, but aren't we getting at this point whenerver the provider is GitHub and the template uses terraform?  If yes, there are, I see the next scenarions:
1) azd pipeline config --auth-type FICS                       :  This is caugh above and returns an error
2) azd pipeline config                                                 :  (No args). This defaults to FICS. So we show the warning about it's not supported and change it to ClientCredentials
3) azd pipeline config --auth-type client-credentials:  Use has choosen to use the secret client credentials. However, we are showing the warning anyway here.

Wouldn't we need an extra `if` here like  `if authType != AuthTypeClientCredentials {`  . So the warning is only showed when client credentials is not set

Reply:
For #1.  Will only fail if CI provider is Azdo or provision provider is terraform
For #2. Will only show warning in Github when provision provider is terraform
For #3. Shouldn't show any error or warnings and work on all provider configurations